zim/gui/widgets.py

```diff
@@ -1998,7 +1998,7 @@ def update_completion(self):
 
 		text = self.get_text()
 		if self._current_completion:
-			if text.startswith(self._current_completion) \
```

Comment:
Well, then, I suppose there is something high-level about this check that is wrong, then. As I read it now, it seems to say "if the user added a non-colon, then we don't have to update the list of completions"... which is clearly incorrect, as (at the very least) adding characters is **expected** to further constrain the list of candidates.